### ISSUE TITLE
Twilio two way SMS through livechat

### ISSUE THREAD

**Issue by designgurudotorg** [RocketChat/Rocket.Chat#6717]
I just setup Twilio credentials (at Administration > SMS) for a Messaging Service I setup within my Twilio account and noticed that sms messages received by my Twilio number were being received in Livechat fine, but that when I responded in Livechat/RocketChat the messages weren't being passed through Twilio (they don't show as outgoing messages at https://www.twilio.com/console/sms/logs)and didn't get received by the original SMS number.

What am I missing?  The only documentation of Twilio/SMS I've found is on StackOverflow (http://stackoverflow.com/questions/42296159/rocketchat-how-to-send-sms-messages-to-livechat-channel-from-mobile) and it doesn't mention anything about configuring outbound settings on twilio.

q./

**Comment by designgurudotorg** [comment]
@sampaiodiego have you guys seen the twilio implementation allow 2 way sms through livechat?  Don't think I've configured anything wrong so I'm not sure if this s a bug with RocketChat or something on Twilio's side to edit.

Here are the steps I did to get Twilio working with RC:
1. Got a new number from Twilio
2. Filled in the RC API url as attached (of course using my appropriate URL)
3. In RC from Administration > SMS, I selected 'Enabled' chose the Twilio service and then filled out the Twilio settings with my Account SID and Auth Token

To recap - when someone texts my Twilio number the text will show up as a new LiveChat, with their phone number as their ID in the left sidebar.  I can receive texts from them in the chat but when I reply Twilio doesn't seem to receive the reply and in turn it can't pass it onto the person texting me.

![screen shot 2017-04-20 at 10 37 11 am](https://cloud.githubusercontent.com/assets/12697545/25236256/68f70c34-25b5-11e7-8013-0b16958f401b.png)

**Comment by designgurudotorg** [comment]
OMG, I had typed in my Twilio Account SID wrong!  That was the only problem going on, once I fixed that I immediately saw 2 way SMS working fine.

This feature is a *game changer* :)

**Comment by MangNPB** [comment]
I'm not very good in coding but have a simple idea for programmers with code blocked.

how to start (initiate) a sms. (currently rocket chat waits for an incoming post with to, from, body... then create a room and.. so on to for us to reply).
simpliest way to initiate this is to have a small script to take a input number and once hit "send sms"  to generate a local post through the rocketchat rest api port with the from:"number just input". this will cause rocket chat to create a new room as if that number just sent us a message and waiting for reply.

---->>> modifying the visitor contact doesn't work!

come on programmer. i need this feature!

Thanks first.  Mang

**Comment by itsarectangle** [comment]
> 
> 
> I'm not very good in coding but have a simple idea for programmers with code blocked.
> 
> how to start (initiate) a sms. (currently rocket chat waits for an incoming post with to, from, body... then create a room and.. so on to for us to reply). simpliest way to initiate this is to have a small script to take a input number and once hit "send sms" to generate a local post through the rocketchat rest api port with the from:"number just input". this will cause rocket chat to create a new room as if that number just sent us a message and waiting for reply.
> 
> ---->>> modifying the visitor contact doesn't work!
> 
> come on programmer. i need this feature!
> 
> Thanks first. Mang

Did you find a way to originate SMS by chance or have a workaround? TIA
